x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/transport/CrossClusterAccessServerTransportFilter.java

```diff
@@ -34,6 +46,39 @@ final class CrossClusterAccessServerTransportFilter extends ServerTransportFilte
         ALLOWED_TRANSPORT_HEADERS = Set.copyOf(allowedHeaders);
     }
 
+    // package private for testing
+    static final Set<String> CROSS_CLUSTER_ACCESS_ACTION_ALLOWLIST;
+    static {
+        final Stream<String> actions = Stream.of(
+            REMOTE_CLUSTER_HANDSHAKE_ACTION_NAME,
+            RemoteClusterNodesAction.NAME,
+            SearchAction.NAME,
+            ClusterStateAction.NAME,
+            ClusterSearchShardsAction.NAME,
+            SearchTransportService.FREE_CONTEXT_SCROLL_ACTION_NAME,
+            SearchTransportService.FREE_CONTEXT_ACTION_NAME,
+            SearchTransportService.CLEAR_SCROLL_CONTEXTS_ACTION_NAME,
+            SearchTransportService.DFS_ACTION_NAME,
+            SearchTransportService.QUERY_ACTION_NAME,
+            SearchTransportService.QUERY_ID_ACTION_NAME,
+            SearchTransportService.QUERY_SCROLL_ACTION_NAME,
+            SearchTransportService.QUERY_FETCH_SCROLL_ACTION_NAME,
+            SearchTransportService.FETCH_ID_SCROLL_ACTION_NAME,
+            SearchTransportService.FETCH_ID_ACTION_NAME,
+            SearchTransportService.QUERY_CAN_MATCH_NAME,
+            SearchTransportService.QUERY_CAN_MATCH_NODE_NAME,
+            TransportOpenPointInTimeAction.OPEN_SHARD_READER_CONTEXT_NAME,
+            ResolveIndexAction.NAME,
+            FieldCapabilitiesAction.NAME,
+            FieldCapabilitiesAction.NAME + "[n]",
+            "indices:data/read/eql"
+        );
+        CROSS_CLUSTER_ACCESS_ACTION_ALLOWLIST = actions
+            // Include action, and proxy equivalent (i.e., with proxy action prefix)
+            .flatMap(name -> Stream.of(name, TransportActionProxy.getProxyAction(name)))
+            .collect(Collectors.toUnmodifiableSet());
```

Comment:
Nit: Not every action has proxy version, e.g. `ClusterStateAction`. But it's a very minor issue since the action will be ultimately checked by privileges.

Reply:
Oh ya I was wondering about that one as well; I've pushed https://github.com/elastic/elasticsearch/pull/94671/commits/f2a13cfd10f20310eb138a1b8b28e110bec96b44 to separate these.